prime-router/src/main/kotlin/fhirengine/engine/QueueMessage.kt

```diff
@@ -85,33 +91,67 @@ data class RawSubmission(
     override val blobSubFolderName: String,
     override val topic: Topic,
     val schemaName: String = "",
-) : Message()
+) : UniversalPipelineQueueMessage()
 
 @JsonTypeName("convert")
-data class FhirConvertMessage(
+data class FhirConvertQueueMessage(
     override val reportId: ReportId,
     override val blobURL: String,
     override val digest: String,
     override val blobSubFolderName: String,
     override val topic: Topic,
     val schemaName: String = "",
-) : Message()
+) : UniversalPipelineQueueMessage()
 
 @JsonTypeName("route")
-data class FhirRouteMessage(
+data class FhirRouteQueueMessage(
     override val reportId: ReportId,
     override val blobURL: String,
     override val digest: String,
     override val blobSubFolderName: String,
     override val topic: Topic,
-) : Message()
+) : UniversalPipelineQueueMessage()
 
 @JsonTypeName("translate")
-data class FhirTranslateMessage(
+data class FhirTranslateQueueMessage(
     override val reportId: ReportId,
     override val blobURL: String,
     override val digest: String,
     override val blobSubFolderName: String,
     override val topic: Topic,
     val receiverFullName: String,
-) : Message()
\ No newline at end of file
+) : UniversalPipelineQueueMessage()
+
+abstract class MixedPipelineQueueMessage : QueueMessage() {
+    abstract val eventType: String
+    abstract val eventAction: Event.EventAction
+}
+
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+data class BatchEventQueueMessage(
```

Comment:
This strikes me as a good opportunity to split up the batch events similarly to how we now have two dedicated queues.  This would allow us to update the UP batching without having to worry about keeping it backwards compatible.

Reply:
I think I actually need to leave this. Where we were looking at was where it splits into sending the event. The event is the same though, it is the message that is different and that is not so easily split up.

Reply:
I'm not necessarily following...the fields are different?

Either way, I'm fine doing it as tech debt.

Reply:
No, what I am saying is that we looked at the code that you thought would be where I would use one Message type vs the other and that looked simple enough, but that wasn't actually where they would need to be split up and where they would need to be split up is much more complicated.